sympy/solvers/ode.py

```diff
@@ -4003,6 +4113,50 @@ def ode_2nd_power_series_regular(eq, func, order, match):
             return Eq(f(x), collect(finalseries1 + finalseries2,
                 [C0, C1]) + Order(x**terms))
 
+def ode_2nd_linear_bessel(eq, func, order, match):
+    r"""
+    Gives solution of the Bessel differential equation
+
+    .. math :: x**2*\frac{d^2y}{dx^2} + x*\frac{dy}{dx}*y(x) + (x**2-n**2)*y(x)
+
+    if n is integer then the solution is of the form Eq(f(x), C0*besselj(n,x)
+    + C1*bessely(n,x)) as both the solutions are linearly independent else if
+    n is a fraction then the solution is of the form Eq(f(x), C0*besselj(n,x)
+    + C1*besselj(-n,x)) which can also transform into Eq(f(x), C0*besselj(n,x)
+    + C1*bessely(n,x)).
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, a
+    >>> from sympy import Symbol
+    >>> v = Symbol('v', positive=True)
+    >>> from sympy.solvers.ode import dsolve, checkodesol
+    >>> from sympy import pprint, Function
+    >>> f = Function('f')
+    >>> y = f(x)
+    >>> genform = x**2*y.diff(x, 2) + x*y.diff(x) + (x**2 - v**2)*y
+    >>> dsolve(genform)
+    Eq(f(x), C1*besselj(v, x) + C2*bessely(v, x))
+
+    References
+    ==========
+
+    https://www.math24.net/bessel-differential-equation/
+
+    """
+    x = func.args[0]
+    f = func.func
+    C0, C1 = get_numbered_constants(eq, num=2)
+    n = match['n']
+    a4 = match['a4']
+    c4 = match['c4']
+    d4 = match['d4']
+    b4 = match['b4']
+    n = sqrt(n**2 + Rational(1, 4)*(c4 - 1)**2)
+    return Eq(f(x), ((x**(Rational(1-c4,2)))*(C0*besselj(S(n)/d4,a4*x**d4/S(d4))
+           + C1*bessely(S(n)/d4,a4*x**d4/S(d4)))).subs(x, x-b4))
```

Comment:
Why are you using `S` here?

Reply:
I don't want to evaluate `n/d4`. I want to make it in the form of Rational.

Reply:
I don't understand. If n is already a SymPy object then S(n) should just return n so it shouldn't have any effect.

Reply:
```
In [3]: S(1)/2
Out[3]: 1/2

In [4]: 1/2
Out[4]: 0.5
```

Reply:
```julia
In [9]: n = sqrt(2)                                                                                                               

In [10]: n/2                                                                                                                      
Out[10]: 
√2
──
2 

In [11]: S(n)/2                                                                                                                   
Out[11]: 
√2
──
2
```
If n is already a SymPy object then it doesn't need to be resympified.

Reply:
got it